jwst/lib/set_telescope_pointing.py

```diff
@@ -3001,3 +3206,105 @@ def cart_to_vector(coord):
     ])
 
     return vector
+
+
+def pa_to_roll_ref(pa: float, siaf: SIAF):
+    """Calculate Roll from the position angle of the given aperture.
+
+    Parameters
+    ----------
+    pa : float
+        Position angle of the aperture, in degrees.
+
+    siaf : SIAF
+        The SIAF of the aperturn
+
+    Returns
+    -------
+    roll_ref : float
+        The roll reference, in degrees
+    """
+    return pa - siaf.v3yangle
+
+
+def t_pars_from_model(model, **t_pars_kwargs):
```

Comment:
The re-work of the TransformParameters will take some studying to see how it might impact the SDP moving target code.

Reply:
This hopefully will not impact anything. It is more a refactor of how `update_wcs_from_telemetry` sets up the initial values in TransformParameters. Ideally, if you are doing something similar, may make things easier.

Reply:
Yeah, that's how it seems to me. The impact might be me deciding to refactor on the SDP side.